ksp_plugin_adapter/plotter.cs

```diff
@@ -0,0 +1,243 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+namespace principia {
+namespace ksp_plugin_adapter {
+
+class Plotter {
+  public Plotter(PrincipiaPluginAdapter adapter) {
+    adapter_ = adapter;
+  }
+
+  public void Plot(DisposablePlanetarium planetarium,
+                   string main_vessel_guid,
+                   double history_length) {
+    PlotCelestialTrajectories(planetarium, main_vessel_guid, history_length);
+    if (main_vessel_guid == null) {
+      return;
+    }
+    PlotVesselTrajectories(planetarium, main_vessel_guid, history_length);
+  }
+
+  private void PlotVesselTrajectories(DisposablePlanetarium planetarium,
+                                      string main_vessel_guid,
+                                      double history_length) {
+    planetarium.PlanetariumPlotPsychohistory(
+        Plugin,
+        main_vessel_guid,
+        history_length,
+        VertexBuffer.data,
+        VertexBuffer.size,
+        out vertex_count_);
```

Comment:
It's odd for `vertex_count_` to be a member variable when it's filled here and consumed at the next line.  I call this the "turd under a stone" antipattern.  Just use a local variable, and pass it explicitly to `DrawLineMesh`.